Riot/Modules/Spaces/SpaceRoomList/ExploreRoom/SpaceExploreRoomViewModel.swift

```diff
@@ -191,6 +235,156 @@ final class SpaceExploreRoomViewModel: SpaceExploreRoomViewModelType {
+                    return SpaceExploreRoomListItemViewData(childInfo: childInfo, avatarViewData: item.avatarViewData)
+                })
+                self.update(viewState: .loaded(self.filteredItemDataList, self.nextBatch != nil && (self.searchKeyword ?? "").isEmpty))
+            case .failure(let error):
+                self.update(viewState: .error(error))
+            }
+        }
+    }
+    
+    private func removeChild(withRoomId roomId: String) {
+        guard let space = session.spaceService.getSpace(withId: spaceId) else {
+            return
+        }
+        
+        self.update(viewState: .loading)
+        space.removeChild(roomId: roomId) { [weak self] response in
+            guard let self = self else { return }
+            
+            switch response {
+            case .success:
+                self.itemDataList = self.itemDataList.filter { $0.childInfo.childRoomId != roomId }
+                self.update(viewState: .loaded(self.filteredItemDataList, self.nextBatch != nil && (self.searchKeyword ?? "").isEmpty))
+            case .failure(let error):
+                self.update(viewState: .error(error))
+            }
+        }
+    }
+    
+    private func joinRoom(withRoomId roomId: String) {
+        self.update(viewState: .loading)
+        self.session.joinRoom(roomId) { [weak self] response in
+            guard let self = self else { return }
+            switch response {
+            case .success:
+                self.update(viewState: .loaded(self.filteredItemDataList, self.nextBatch != nil && (self.searchKeyword ?? "").isEmpty))
+            case .failure(let error):
+                self.update(viewState: .error(error))
+            }
+        }
+    }
+
+    
+    // MARK: - ContextMenu
+    
+    @available(iOS 13.0, *)
+    private func contextMenu(forRoom itemData: SpaceExploreRoomListItemViewData, isJoined: Bool, canSendSpaceStateEvents: Bool) -> UIMenu {
+        return UIMenu(children: [
+            inviteAction(for: itemData, isJoined: isJoined),
+            suggestAction(for: itemData, canSendSpaceStateEvents: canSendSpaceStateEvents),
+            isJoined ? settingsAction(for: itemData, isJoined: isJoined) :joinAction(for: itemData, isJoined: isJoined),
+            removeAction(for: itemData, canSendSpaceStateEvents: canSendSpaceStateEvents)
+        ])
+    }
+    
+    @available(iOS 13.0, *)
+    private func contextMenu(forSpace itemData: SpaceExploreRoomListItemViewData, isJoined: Bool, canSendSpaceStateEvents: Bool) -> UIMenu {
+        return UIMenu(children: [
+            inviteAction(for: itemData, isJoined: isJoined),
+            suggestAction(for: itemData, canSendSpaceStateEvents: canSendSpaceStateEvents),
+            isJoined ? settingsAction(for: itemData, isJoined: isJoined) :joinAction(for: itemData, isJoined: isJoined),
+            removeAction(for: itemData, canSendSpaceStateEvents: canSendSpaceStateEvents)
```

Comment:
Seeing as the menu contents appears to be identical (to my eyes) I'm assuming that having a `forRoom` and `forSpace` methods is for future work?

Reply:
I made those 2 methods for future work indeed but I can merge them into one method if you prefer.

Reply:
No no, was only double checking in case it wasn't, it's totally fine as is 🙂